spectator-ext-aws/src/main/java/com/netflix/spectator/aws/SpectatorServiceMetricCollector.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2015 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spectator.aws;
+
+import com.amazonaws.metrics.ByteThroughputProvider;
+import com.amazonaws.metrics.ServiceLatencyProvider;
+import com.amazonaws.metrics.ServiceMetricCollector;
+import com.netflix.spectator.api.Id;
+import com.netflix.spectator.api.Registry;
+import com.netflix.spectator.impl.Preconditions;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * A ServiceMetricCollector that captures service metrics in a Registry.
+ */
+public class SpectatorServiceMetricCollector extends ServiceMetricCollector {
+    private static final long NANOS_PER_MILLI = TimeUnit.MILLISECONDS.toNanos(1);
+    private final Registry registry;
+
+    /**
+     * Constructs a new SpectatorServiceMetricCollector.
+     *
+     * @param registry the Registry for this SpectatorServiceMetricCollector.
+     */
+    public SpectatorServiceMetricCollector(Registry registry) {
+        super();
+        this.registry = Preconditions.checkNotNull(registry, "registry");
+    }
+
+    @Override
+    public void collectByteThroughput(ByteThroughputProvider provider) {
+        if (provider.getByteCount() > 0) {
+            registry.counter(getThroughputBytesId(provider)).increment(provider.getByteCount());
+            registry.timer(getThroughputDurationId(provider)).record(provider.getDurationNano(), TimeUnit.NANOSECONDS);
```

Comment:
I'm actually not even sure how valid this whole class is - the providers passed as collect arguments don't seem to expose enough info to add enough tags for these metrics.

For example: there is nothing indicating the region for the service, so collecting these is just going to be a mashup of the service latency to each region (maybe useful if you only talk inside a region, not so much if you have a service that makes calls to multiple regions)

The provider's all have a providerId attribute but I dropped that from the tags since it turned out to just be the default Object::toString value